pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -462,11 +462,22 @@ func (r *ReconcilePerconaServerMongoDB) reconcileStatefulSet(arbiter bool, cr *a
 	if err != nil {
 		return nil, fmt.Errorf("create StatefulSet.Spec %s: %v", sfs.Name, err)
 	}
-	sfsSpec.Template.Annotations = sfs.Spec.Template.Annotations
+
 	if sfsSpec.Template.Annotations == nil {
 		sfsSpec.Template.Annotations = make(map[string]string)
 	}
 
+	if sfs.Annotations == nil {
+		sfs.Annotations = make(map[string]string)
+	}
+
+	for k, v := range sfs.Spec.Template.Annotations {
+		sfsSpec.Template.Annotations[k] = v
+		if !strings.HasPrefix(k, "percona.com") {
+			sfs.Annotations[k] = v
+		}
+	}
+
```

Comment:
why we filter percona annotations here?

Reply:
I don't think that ssl fields should be placed into statefulset annotation

Reply:
but we add them intentionally, we need restart all pods if TLS certificates are changed

Reply:
upgraded